wai-aria/role/grid-roles.html

```diff
@@ -0,0 +1,53 @@
+<!doctype html>
+<html>
+  <head>
+    <title>Grid Role Verification Tests</title>
+    <script src="/resources/testharness.js"></script>
+    <script src="/resources/testharnessreport.js"></script>
+    <script src="/resources/testdriver.js"></script>
+    <script src="/resources/testdriver-vendor.js"></script>
+    <script src="/resources/testdriver-actions.js"></script>
+    <script src="/wai-aria/scripts/aria-utils.js"></script>
+  </head>
+  <body>
+
+  <p>Tests <a href="https://w3c.github.io/aria/#grid">grid</a> and related roles.</p>
+
+  <div role="grid" data-testname="role is grid" data-expectedrole="grid" class="ex">
+    <div role="row" data-testname="role is row (in grid)" data-expectedrole="row" class="ex">
+      <span role="columnheader" data-testname="role is columnheader (in row, in grid)" data-expectedrole="columnheader" class="ex">x</span>
+      <span role="columnheader">x</span>
+    </div>
+    <div role="row">
+      <span role="rowheader" data-testname="role is rowheader (in row, in grid)" data-expectedrole="rowheader" class="ex">x</span>
+      <span role="gridcell" data-testname="role is gridcell (in row, in grid)" data-expectedrole="gridcell" class="ex">x</span>
+    </div>
+  </div>
+
+  <div role="grid">
+    <div role="rowgroup" data-testname="role is rowgroup (in grid)" data-expectedrole="rowgroup" class="ex">
+      <div role="row" data-testname="role is row (in rowgroup, in grid)" data-expectedrole="row" class="ex">
+        <span role="columnheader" data-testname="role is columnheader (in row, in rowgroup, in grid)" data-expectedrole="columnheader" class="ex">x</span>
+        <span role="columnheader">x</span>
+        <span role="columnheader">x</span>
+      </div>
+    </div>
+    <div role="rowgroup">
+      <div role="row">
+        <span role="rowheader" data-testname="role is rowheader (in row, in rowgroup, in grid)" data-expectedrole="rowheader" class="ex">x</span>
+        <span role="gridcell" data-testname="role is gridcell (in row, in rowgroup, in grid)" data-expectedrole="gridcell" class="ex">x</span>
+        <span role="gridcell">x</span>
+      </div>
+    </div>
+  </div>
+
+  <div role="gridcell" data-testname="should not have role gridcell outside of row" data-expectedrole="none" class="ex">x</div>
+  <div role="rowheader" data-testname="should not have role rowheader outside of row" data-expectedrole="none" class="ex">x</div>
+  <div role="columnheader" data-testname="should not have role columnheader outside of row" data-expectedrole="none" class="ex">x</div>
```

Comment:
I wonder if div fallback to role "none" is the most reliable test. Commenting here as an action for me to check the CI results on this one. If it differs per browser, that issue is exercised in a different test.

Reply:
Update: Looks like all 3 fail with the same error, so I think this is okay as-is? We can adjust later something problematic is discovered in implementation. 

@howard-e Will you file the respective bugs in Chromium, Gecko, and WebKit once this lands?

Reply:
> Update: Looks like all 3 fail with the same error, so I think this is okay as-is? We can adjust later something problematic is discovered in implementation.

Okay, thanks for checking that. Sounds good

> @howard-e Will you file the respective bugs in Chromium, Gecko, and WebKit once this lands?

@cookiecrook sure, I'm fine with doing that

Reply:
FWIW, I usually wait to file the implementation bugs until there is one published result from the PR, so the same results link can be used for all of the engines. I recall sometimes it takes a day or two after the PR lands.

Reply:
Bugs filed at:

* https://bugzilla.mozilla.org/show_bug.cgi?id=1849160
* https://bugs.chromium.org/p/chromium/issues/detail?id=1473647
* https://bugs.webkit.org/show_bug.cgi?id=260348